docs/cudf/source/developer_guide/benchmarking.md

```diff
@@ -0,0 +1,288 @@
+# Benchmarking cuDF
+
+The goal of the benchmarks in this repository is to measure the performance of various cuDF APIs.
+Benchmarks in cuDF are written using the
+[`pytest-benchmark`](https://pytest-benchmark.readthedocs.io/en/latest/index.html) plugin to the
+[`pytest`](https://docs.pytest.org/en/latest/) Python testing framework.
+Using `pytest-benchmark` provides a seamless experience for developers familiar with `pytest`.
+We include benchmarks of both public APIs and internal functions.
+The former give us a macro view of our performance, especially vis a vis pandas.
+The latter help us quantify and minimize the overhead of our Python bindings.
+
+```{note}
+Our current benchmarks focus entirely on measuring run time.
+However, minimizing memory footprint can be just as important for some cases.
+In the future, we may update our benchmarks to also include memory usage measurements.
+```
+
+## Benchmark organization
+
+At the top level benchmarks are divided into `internal` and `API` directories.
+API benchmarks are for public features that we expect users to consume.
+Internal benchmarks capture the performance of cuDF internals that have no stability guarantees.
+
+Within each directory, benchmarks are organized based on the type of function.
+Functions in cuDF generally fall into two groups:
+
+1. Methods of classes like `DataFrame` or `Series`.
+2. Free functions operating on the above classes like `cudf.merge`.
+
+The former should be organized into files named `bench_class.py`.
+For example, benchmarks of `DataFrame.eval` belong in `API/bench_dataframe.py`.
+Benchmarks should be written at the highest level of generality possible with respect to the class hierarchy.
+For instance, all classes support the `take` method, so those benchmarks belong in `API/bench_frame_or_index.py`.
+If a method has a slightly different API for different classes, benchmarks should use a minimal common API,
+_unless_ developers expect certain arguments to trigger code paths with very different performance characteristics.
```

Comment:
Are there any examples we could include here?

Reply:
Good idea. I added a mention of `where`. I didn't try to link since the line numbers will change frequently with the addition of new benchmarks, but the mention should be sufficient.